src/opm/output/eclipse/AggregateConnectionData.cpp

```diff
@@ -54,26 +55,40 @@ namespace {
     }
 
     template <class ConnOp>
-    void connectionLoop(const std::vector<Opm::Well>& wells,
-                        const Opm::EclipseGrid&       grid,
-                        const Opm::data::WellRates&   xw,
-                        ConnOp&&                      connOp)
+    void connectionLoop(const Opm::EclipseGrid& grid,
+                        const Opm::Well&        well,
+                        const std::size_t       wellID,
+                        const Opm::data::Well*  wellRes,
+                        ConnOp&&                connOp)
     {
-        for (auto nWell = wells.size(), wellID = 0*nWell;
-             wellID < nWell; ++wellID)
-        {
-            const auto& well = wells[wellID];
-            const auto well_iter = xw.find(well.name());
-            const Opm::data::Well * well_rates = (well_iter == xw.end()) ? nullptr : &well_iter->second;
-            const auto& connections = well.getConnections().output(grid);
-            std::size_t connID = 0;
-            for (const auto& conn : connections) {
-                if (well_rates)
-                    connOp(wellID, *conn, connID, well_rates->find_connection(conn->global_index()));
-                else
-                    connOp(wellID, *conn, connID, nullptr);
-                connID++;
-            }
+        std::size_t connID = 0;
+        for (const auto* connPtr : well.getConnections().output(grid)) {
+            const auto* dynConnRes = (wellRes == nullptr)
+                ? nullptr : wellRes->find_connection(connPtr->global_index());
+
+            connOp(wellID, *connPtr, connID, dynConnRes);
+
+            ++connID;
+        }
+    }
+
+    template <class ConnOp>
+    void connectionLoop(const Opm::Schedule&        sched,
```

Comment:
This looks like it could be named something like `wellConnectionLoop`?

Reply:
Done.

Reply:
:+1: